frontend/pages/signup.tsx

```diff
@@ -22,16 +21,17 @@ import { Container, Theme, useMediaQuery } from "@mui/material";
 import getHubTheme from "../src/themes/fetchHubTheme";
 import { transformThemeData } from "../src/themes/transformThemeData";
 import CustomAuthImage from "../src/components/hub/CustomAuthImage";
+import { extractHubFromRedirectUrl } from "../public/lib/hubOperations";
 
 export async function getServerSideProps(ctx) {
-  const hubUrl = ctx.query.hub;
-
-  const hubThemeData = await getHubTheme(hubUrl);
+  const { redirect } = ctx.query;
+  const hubUrl = extractHubFromRedirectUrl(redirect);
+  const hubThemeData = hubUrl ? await getHubTheme(hubUrl) : null;
```

Comment:
this breaks the normal link without a redirect.
```
http://localhost:3000/de/signin?hub=prio1
```
Additionally, when trying to access the login page via the nav bar from the hub-browse page, it uses 
```
http://localhost:3000/de/signin?redirect=%2Fhubs%2Fprio1&hub=prio1
```
and breaks, as it is comming from `/de/hubs/prio1` which does not contain a url param.